File: Difficulty: Medium/Validate an IP Address/validate-an-ip-address.py
#User function Template for python3
class Solution:
    def isValid(self, str):
        #code here
        ip = str.split(".")
        
        if len(ip) != 4:
            return False
            
        for i in ip:
            
            # print(len(i))
            
            if (len(i) > 1 and i[0] == "0"):
                return False
                
            if (i == ""):
                return False
            
            ip_num = int(i)
            
            if ip_num < 0 or ip_num > 255:
                
                # print(ip_num)
                
                return False
                
        return True
                



#{ 
 # Driver Code Starts
#Initial Template for Python 3

if __name__ == "__main__":
    t = int(input())
    for _ in range(0, t):
        s = input()
        ob = Solution()
        if (ob.isValid(s)):
            print("true")
        else:
            print("false")

# } Driver Code Ends

File: Difficulty: Medium/Validate an IP Address/validate-an-ip-address.java
//{ Driver Code Starts
// Initial Template for Java

// Initial Template for Java

import java.io.*;
import java.util.*;

public class validip {

    public static void main(String[] args) {

        Scanner sc = new Scanner(System.in);
        int t = sc.nextInt();

        while (t-- > 0) {
            String s = sc.next();
            Solution obj = new Solution();

            if (obj.isValid(s))
                System.out.println("true");
            else
                System.out.println("false");

            System.out.println("~");
        }
    }
}
// } Driver Code Ends


// User function Template for Java

class Solution {

    public boolean isValid(String s) {
        // Write your code here
        
        String[] str = s.split("[\\.]");
        
        if (str.length != 4) {
            return false;
        }
        
        for (String str1 : str) {
            
            if (str1.length() > 1 && str1.charAt(0) == '0') return false;
            
            if (str1.equals("")) return false;
            
            if (Integer.parseInt(str1) > 255 || Integer.parseInt(str1) < 0) {
                return false;
            }
            
            
        }
        
        return true;
    }
}